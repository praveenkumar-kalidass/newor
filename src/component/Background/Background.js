import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Polygon, Svg } from 'react-native-svg';

import COLOR from '../../constant/color';
import BackgroundContainer from './Background.style';

const Background = ({ height, width }) => {
  const primaryBackground = useMemo(
    () => `0,0 0,${height} ${width},${height} 30,${height - 30} 0,0`,
    [height, width],
  );
  const secondaryBackground = useMemo(
    () => `0,0 ${width},0 ${width},${height} ${width - 30},30 0,0`,
    [height, width],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Polygon fill={COLOR.LIGHT_PRIMARY_100} points={primaryBackground} />
        <Polygon fill={COLOR.LIGHT_SECONDARY_100} points={secondaryBackground} />
      </Svg>
    </BackgroundContainer>
  );
};

Background.propTypes = {
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
};

export default Background;
