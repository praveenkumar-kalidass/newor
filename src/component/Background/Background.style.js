import styled from 'styled-components/native';

export default styled.View`
  background-color: ${({ theme }) => theme.color.BACKGROUND_100};
  position: absolute;
`;
