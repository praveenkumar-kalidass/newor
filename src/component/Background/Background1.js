import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Polygon, Svg } from 'react-native-svg';

import useTheme from 'theme/useTheme';
import BackgroundContainer from './Background.style';

const Background1 = ({ height, width }) => {
  const theme = useTheme();
  const primaryBackground = useMemo(
    () => `0,0 0,${height} ${width},${height} 30,${height - 30} 0,0`,
    [],
  );
  const secondaryBackground = useMemo(
    () => `0,0 ${width},0 ${width},${height} ${width - 30},30 0,0`,
    [],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Polygon fill={theme.color.PRIMARY_100} points={primaryBackground} />
        <Polygon fill={theme.color.SECONDARY_100} points={secondaryBackground} />
      </Svg>
    </BackgroundContainer>
  );
};

Background1.propTypes = {
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
};

export default Background1;
