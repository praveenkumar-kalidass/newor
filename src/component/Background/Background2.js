import React, { useMemo } from 'react';
import { useWindowDimensions } from 'react-native';
import Svg, { Path } from 'react-native-svg';

import COLOR from '../../constant/color';
import BackgroundContainer from './Background.style';

const Background2 = () => {
  const { height, width } = useWindowDimensions();

  const primaryDataPoint = useMemo(
    () => `M0 0 L0 ${height / 2} L${width / 2} 0 L0 0 Z`,
    [],
  );

  const secondaryDataPoint = useMemo(
    () => `M${width} ${height} L${width / 2} ${height} L${width} ${height / 2} L${width} ${height} Z`,
    [],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Path d={primaryDataPoint} fill={COLOR.LIGHT_PRIMARY_100} />
        <Path d={secondaryDataPoint} fill={COLOR.LIGHT_SECONDARY_100} />
      </Svg>
    </BackgroundContainer>
  );
};

export default Background2;
