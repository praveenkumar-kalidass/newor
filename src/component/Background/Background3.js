import React, { useMemo } from 'react';
import { useWindowDimensions } from 'react-native';
import { Path, Svg } from 'react-native-svg';

import useTheme from 'theme/useTheme';
import BackgroundContainer from './Background.style';

const Background3 = () => {
  const { height, width } = useWindowDimensions();
  const theme = useTheme();

  const dataPoint = useMemo(
    () => `M 0,0
      L 0,${height / 5}
      C 50 ${(height / 5) + 50}, ${width - 50} ${(height / 5) + 50}, ${width} ${height / 5}
      L ${width},0
      Z`,
    [],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Path d={dataPoint} fill={theme.color.SECONDARY_100} />
      </Svg>
    </BackgroundContainer>
  );
};

export default Background3;
