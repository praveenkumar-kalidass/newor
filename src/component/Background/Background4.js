import React from 'react';
import { useWindowDimensions } from 'react-native';
import { Circle, Svg } from 'react-native-svg';

import useTheme from 'theme/useTheme';
import BackgroundContainer from './Background.style';

const Background4 = () => {
  const { height, width } = useWindowDimensions();
  const theme = useTheme();

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Circle
          cx={0}
          cy={height}
          r={250}
          stroke={theme.color.SECONDARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
        <Circle
          cx={0}
          cy={height}
          r={150}
          stroke={theme.color.SECONDARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
        <Circle
          cx={width}
          cy={0}
          r={250}
          stroke={theme.color.PRIMARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
        <Circle
          cx={width}
          cy={0}
          r={150}
          stroke={theme.color.PRIMARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
      </Svg>
    </BackgroundContainer>
  );
};

export default Background4;
