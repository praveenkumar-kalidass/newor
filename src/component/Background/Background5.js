import React from 'react';
import { useWindowDimensions } from 'react-native';
import { Svg, ForeignObject } from 'react-native-svg';
import { Text } from 'native-base';

import CONSTANT from 'constant';
import useTheme from 'theme/useTheme';
import BackgroundContainer from './Background.style';

const Background5 = () => {
  const { height, width } = useWindowDimensions();
  const theme = useTheme();

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <ForeignObject x={0} y={height - 350}>
          <Text color={theme.color.PRIMARY_100} fontSize={200}>
            {CONSTANT.APP_LITERAL.RUPEE_SYMBOL}
          </Text>
        </ForeignObject>
        <ForeignObject x={width - 225} y={-100}>
          <Text color={theme.color.SECONDARY_100} fontSize={400}>
            {CONSTANT.APP_LITERAL.RUPEE_SYMBOL}
          </Text>
        </ForeignObject>
      </Svg>
    </BackgroundContainer>
  );
};

export default Background5;
