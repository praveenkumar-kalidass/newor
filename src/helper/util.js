import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
