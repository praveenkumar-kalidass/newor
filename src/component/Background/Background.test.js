import React from 'react';

import { render } from '../../test/util';
import Background from './Background';

describe('Background', () => {
  it('should match snapshot', () => {
    const container = render(<Background />);

    expect(container).toMatchSnapshot();
  });
});
