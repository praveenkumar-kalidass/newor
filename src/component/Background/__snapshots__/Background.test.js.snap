// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "position": "absolute",
        },
      ]
    }
  >
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth="100%"
      focusable={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M0 0 0 undefined undefined undefined 30 NaN 0 0z"
          fill={4289527808}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M0 0 undefined 0 undefined undefined NaN 30 0 0z"
          fill={4278461322}
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</RNCSafeAreaProvider>
`;
